src/elevenlabs/conversation.py

```diff
@@ -0,0 +1,282 @@
+            while not self._should_stop.is_set():
+                try:
+                    message = json.loads(ws.recv(timeout=0.5))
+                    self._handle_message(message, ws)
+                except TimeoutError:
+                    pass
+
+    def _handle_message(self, message, ws):
+        if message["type"] == "conversation_initiation_metadata":
+            event = message["conversation_initiation_metadata_event"]
+            assert self._conversation_id is None
+            self._conversation_id = event["conversation_id"]
+        elif message["type"] == "audio":
+            event = message["audio_event"]
+            if int(event["event_id"]) <= self._last_interrupt_id:
+                return
+            audio = base64.b64decode(event["audio_base_64"])
+            self.audio_interface.output(audio)
+        elif message["type"] == "agent_response":
+            if self.callback_agent_response:
+                event = message["agent_response_event"]
+                self.callback_agent_response(event["agent_response"].strip())
+        elif message["type"] == "user_transcript":
+            if self.callback_user_transcript:
+                event = message["user_transcription_event"]
+                self.callback_user_transcript(event["user_transcript"].strip())
+        elif message["type"] == "interruption":
+            event = message["interruption_event"]
+            self.last_interrupt_id = int(event["event_id"])
+            self.audio_interface.interrupt()
+        elif message["type"] == "ping":
+            event = message["ping_event"]
+            ws.send(
+                json.dumps(
+                    {
+                        "type": "pong",
+                        "event_id": event["event_id"],
+                    }
+                )
+            )
+            if self.callback_latency_measurement and event["ping_ms"]:
+                self.callback_latency_measurement(int(event["ping_ms"]))
+        else:
+            pass  # Ignore all other message types.
+
+    def _get_wss_url(self):
+        base_url = self.client._client_wrapper._base_url
+        # Replace http(s) with ws(s).
+        base_ws_url = base_url.replace("http", "ws", 1)  # First occurrence only.
+        return f"{base_ws_url}/v1/convai/conversation?agent_id={self.agent_id}"
+
+    def _get_signed_url(self):
+        # TODO: Use generated SDK method once available.
+        response = self.client._client_wrapper.httpx_client.request(
+            f"v1/convai/conversation/get_signed_url?agent_id={self.agent_id}",
+            method="GET",
+        )
+        return response.json()["signed_url"]
+
+
+def get_default_audio_interface():
+    """Returns a default audio interface implementation.
+
+    Uses the default system input and output sound devices.
+    """
+    try:
+        import pyaudio
+    except ImportError:
+        raise ImportError("To use the default audio interface, you must install and pyaudio.")
```

Comment:
This could be part of __init__() and users could just directly import Default(Audio)Interface instead of the get_default_audio_interface, no?

Reply:
Sure, updated.